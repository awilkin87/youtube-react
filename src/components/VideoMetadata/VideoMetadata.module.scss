@use '../../styles/shared.module';

.container {
  color: shared.$secondary-text-color;
  font-size: 16px;

  h3 {
    color: shared.$primary-text-color;
    font-weight: 400;
  }
 
  .stats {
    display: flex;
    justify-content: space-between;
    width: 100%;
    align-items: center;

    .actions {
      display: flex;
      align-items: center;

      // Customize Semantic UI button
      :global(.button.labeled) {
        @include shared.labeled-button;
      }
      
      & > *:not(:last-child) {
        margin-right: 8px;
      }
    }
  }
}