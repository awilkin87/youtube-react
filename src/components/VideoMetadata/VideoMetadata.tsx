import { Divider } from 'semantic-ui-react';
import styles from './VideoMetadata.module.scss';

export interface VideoMetaDataProps {
  title: string;
  viewCount: number;
}

const VideoMetaData: React.FunctionComponent<Readonly<VideoMetaDataProps>> = function (props) {
  return (
    <div className={styles.container}>
      <h3>{props.title}</h3>
      <div className={styles.stats}>
        <span>{props.viewCount.toLocaleString()} views</span>
        <div>
          {/* TODO */}
        </div>
      </div>
      <Divider />
    </div>
  );
}

export default VideoMetaData;