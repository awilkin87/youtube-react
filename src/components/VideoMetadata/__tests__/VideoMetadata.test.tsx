import { shallow } from 'enzyme'
import VideoMetadata from '../VideoMetadata'

// This is probably not a good test as the view count output is dependent on locale.
// A dev running the tests in a different locale might get test failures.
describe('VideoMetadata', () => {
  test('Render VideoMetadata with zero views', () => {
    const wrapper = shallow(<VideoMetadata title='Video Title' viewCount={0} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  test('Render VideoMetadata with views', () => {
    const wrapper = shallow(<VideoMetadata title='Video Title' viewCount={1000} />);
    expect(wrapper).toMatchSnapshot();
  });
});
