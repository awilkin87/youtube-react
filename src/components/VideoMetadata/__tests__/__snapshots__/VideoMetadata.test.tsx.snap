// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoMetadata Render VideoMetadata with views 1`] = `
<div
  className="container"
>
  <h3>
    Video Title
  </h3>
  <div
    className="stats"
  >
    <span>
      1,000
       views ᛫ Aug 14, 2017
    </span>
    <div
      className="actions"
    >
      <VideoRating
        dislikes={25}
        likes={75}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="add circle"
        />
        Save
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;

exports[`VideoMetadata Render VideoMetadata with zero views 1`] = `
<div
  className="container"
>
  <h3>
    Video Title
  </h3>
  <div
    className="stats"
  >
    <span>
      0
       views ᛫ Aug 14, 2017
    </span>
    <div
      className="actions"
    >
      <VideoRating
        dislikes={25}
        likes={75}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="add circle"
        />
        Save
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;
